src/3d/qgspointcloudlayer3drenderer.cpp

```diff
@@ -74,6 +76,28 @@ void QgsPointCloud3DRenderContext::cancelRendering() const
 {
   mFeedback->cancel();
 }
+
+void QgsPointCloud3DRenderContext::updateExtent()
+{
+  if ( map().extent().isEmpty() )
+  {
+    // an empty extent means no filter, so let's pass it without transformation
+    mExtent = QgsRectangle();
+  }
+  else
+  {
+    try
+    {
+      mExtent = mCoordinateTransform.transformBoundingBox( map().extent(), Qgis::TransformDirection::Reverse );
+    }
+    catch ( const QgsCsException & )
+    {
+      // bad luck, can't reproject for some reason. Let's use an empty extent to skip filtering.
+      QgsDebugMsg( QStringLiteral( "Transformation of extent failed!" ) );
+      mExtent = QgsRectangle();
+    }
```

Comment:
this extent transform from one crs to another is done a multiple time and could be a member function of QgsRectangle or QgsAABB

Reply:
You mean wrapping up the _try-catch_ block in a function?
The thing is that handling in the case of an exception may be different depending on where it is used. Here, for example, if the extent transformation fails, we want to default to an empty rectangle so that no actual filter is passed to the point cloud provider. In other cases we may default to the un-transformed extent.